Fix LVM resize error due to fs utilities do not support renamed devices

praveenkumar opened the pull request:
As part of disk resize for microshift looks like there is bug in the lvextend tool and it is not work when extending a lvm with resize.

- https://access.redhat.com/solutions/7016718
- https://bugzilla.redhat.com/show_bug.cgi?id=2211856

This PR is a workaround which is provided as part of BZ where `lvextend` shouldn't run with `-r` (resize) option. So now we are using `lvextend` only for extending the respective logical volume and then using `xfs_growfs` to grow that volume.


**Fixes:** Issue #3705

cfergeau commented on pkg/crc/machine/start.go:
```diff
@@ -164,51 +169,52 @@ func runGrowpart(sshRunner *crcssh.Runner, rootPart string) error {
 	vgFree := 16127098880
 	expectedLVSize := vgSize - vgFree
 	sizeToIncrease := expectedLVSize - lvSize
+	lgVol := "/dev/rhel/root"
```
does `lgVol` stand for `logical volume`? It's usually shortened to `lv`, `lvdisplay` calls it `lvPath`.

cfergeau replied:
Instead of hardcoding `/dev/rhel/root` here and returning it back to the caller, I'd pass `rhel/root` as a `lvFullName` argument, and make this method independent of the VG/PV used (lgVol becomes `dev`+`lvFullName`, and the various `-S lv_name=root` can be replaced with `-S lv_full_name=rhel/root`).

anjannath commented:
Tested and the error reported in #3705 doesn't occur with this PR, inside the VM `/sysroot` size is also increased:
```
Filesystem             Size  Used Avail Use% Mounted on

/dev/mapper/rhel-root   24G  4.6G   20G  19% /sysroot
```

cfergeau commented on pkg/crc/machine/start.go:
```diff
@@ -169,8 +169,8 @@ func runGrowpart(sshRunner *crcssh.Runner, rootPart string) error {
 
-func growLVForMicroshift(sshRunner *crcssh.Runner) (string, error) {
-	rootPart, _, err := sshRunner.RunPrivileged("Get PV disk path", "/usr/sbin/pvs", "-S", "lv_name=root", "-o", "pv_name", "--noheadings")
+func growLVForMicroshift(sshRunner *crcssh.Runner, lvFullName string) (string, error) {
```
By passing `lvFullName` as an arg, you no longer need to return `lvPath` from this method, which simplifies the other commit. I'd start the PR with this refactor, and then the commit adding the workaround for the resize.

praveenkumar replied:
@cfergeau so what you want is from the caller something like following?
```golang
if preset == crcPreset.Microshift {
               lvFullName := "rhel/root"
               rootPart = fmt.Sprintf("/dev/%s", lvFullName)
		err = growLVForMicroshift(sshRunner, lvFullName)
		if err != nil {
			return err
		}
	}
```

cfergeau replied:
Yes unless you think it does not make sense? This would be close to the way the other codepath is structured.

praveenkumar replied:
Alright, I will make that change, wanted to make sure I understood correctly.

praveenkumar replied:
@cfergeau updated.

cfergeau replied:
Thanks  for the refactor!

praveenkumar committed:
start/growLVForMicroshift: Use lvFullName as argument

This is just a refactor to avoid using `root` as lv_name and
then `/dev/rhel/root` as lvPath in `growLVForMicroshift` function,
just use the full name of lv and then create the lvPath from it.

praveenkumar committed:
Fix LVM resize error due to fs utilities do not support renamed devices

As part of disk resize for microshift looks like there is bug in the
lvextend tool and it is not work when extending a lvm with resize.

- https://access.redhat.com/solutions/7016718
- https://bugzilla.redhat.com/show_bug.cgi?id=2211856

This PR is a workaround which is provided as part of BZ where `lvextend`
shouldn't run with `-r` (resize) option. So now we are using `lvextend`
only for extending the respective logical volume and then using
`xfs_growfs` to grow that volume.
